### PR TITLE
ENH Only query primary DB

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-framework/issues/11352

### CODE REVIEW

File: src/Model/HybridSessionDataObject.php [hunk 1]
```diff
@@ -20,4 +20,6 @@ class HybridSessionDataObject extends DataObject
     ];
 
     private static $table_name = 'HybridSessionDataObject';
+
+    private static bool $must_use_primary_db = true;
```
Comment (ChloeSartorelli) [inline src/Model/HybridSessionDataObject.php:28]: Why? Add a code comment so it's clear why we're setting this to true. Also add to the `withPrimary()` call in `DatabaseStore`

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: src/Model/HybridSessionDataObject.php (hunk 1 above)
Comment (emteknetnz) [reply src/Model/HybridSessionDataObject.php:28]: Updated

Comment (ChloeSartorelli) [reply src/Model/HybridSessionDataObject.php:28]: The comment explains what this config property _does_ - but still doesn't say why we need it to be true for _this_ dataobject specifically. I assume it's something to do with ensuring session data doesn't go out of sync?

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: src/Model/HybridSessionDataObject.php (hunk 1 above)
Comment (emteknetnz) [reply src/Model/HybridSessionDataObject.php:28]: Updated

**Review by GuySartorelli — APPROVED** [review]
LGTM, will merge after framework PR

**Pull request merged**